modular_skyrat/modules/implants/code/augments_brain.dm

```diff
@@ -0,0 +1,92 @@
+	name = "empathic sensor implant"
+	desc = "This won't fix your social issues, but it may help you repress them better." //not final
+	icon_state = "brain_implant_antidrop"
+	var/active = FALSE
+	var/list/stored_items = list()
+	slot = ORGAN_SLOT_BRAIN_CEREBELLUM
+	var/modifies_speech = TRUE
+	
+// This should appropriately grant and remove empathy from the implant - important if you are a shadekin for some reason to not remove native languages, and allow the procs to work
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/emp_act(severity)
+	. = ..()
+	if((organ_flags & ORGAN_FAILING) || . & EMP_PROTECT_SELF)
+		return
+	
+	owner.adjust_stutter(30 SECONDS)
+	organ_flags |= ORGAN_FAILING
+	UnregisterSignal(owner, COMSIG_MOB_SAY)
+	owner.remove_language(/datum/language/marish/empathy, source = LANGUAGE_IMPLANT)
+	addtimer(CALLBACK(src, PROC_REF(reboot)), 180 / severity)
+	to_chat(owner, span_warning("You feel overwhelmed!"))
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/proc/implant_ready()
+	RegisterSignal(owner, COMSIG_MOB_SAY, PROC_REF(handle_speech))
+	owner.grant_language(/datum/language/marish/empathy, source = LANGUAGE_IMPLANT)
+	to_chat(owner, span_abductor("Your mind opens to others. You can hear the thoughts of those around you, but only faintly."))
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/on_mob_remove(mob/living/carbon/implant_owner)
+	. = ..()
+	organ_flags &= ~ORGAN_FAILING
+	UnregisterSignal(implant_owner, COMSIG_MOB_SAY)
+	implant_owner.remove_language(/datum/language/marish/empathy, source = LANGUAGE_IMPLANT)
+	if(QDELETED(src))
+		return
+	to_chat(implant_owner, span_abductor("Your mind closes from others. It's quiet, now."))
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/on_mob_insert(mob/living/carbon/receiver)
+	. = ..()
+	to_chat(receiver, span_abductor("You begin to feel an awareness of those around you."))
+	addtimer(CALLBACK(src, PROC_REF(implant_ready)), 90)
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/proc/reboot()
+	organ_flags &= ~ORGAN_FAILING
+	implant_ready()
+
+//code mostly lifted from shadekins
+	
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/proc/modify_speech(datum/source, list/speech_args)
+	ASYNC
+		if(organ_flags & ORGAN_FAILING) //just in case they somehow activate this while the implant is disabled...
+			to_chat(owner, span_abductor("You are unable to project your thoughts."))
+			return
+		actually_modify_speech(source, speech_args)
+	speech_args[SPEECH_MESSAGE] = "" // Makes it not send to chat verbally
+	
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/proc/handle_speech(datum/source, list/speech_args)
+	if(speech_args[SPEECH_LANGUAGE] == /datum/language/marish/empathy)
+		return modify_speech(source, speech_args)
+
+//I couldnt get this to work as a TYPE_PROC_REF. So it's copied.
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/proc/actually_modify_speech(datum/source, list/speech_args)
```

Comment:
I'd like to see if other maintainers have ideas how to do that, it could be as simple as adding a fake organ set inside the implant

Reply:
Since now they're different, marking this as resolved unless we want to modularize the ability.